### PR TITLE
Add nodeSelector and affinity to support dedicated nodes

### PR DESCRIPTION
**Pull request by emmanuelstroem**
### Description

In a situation where the runner has a dedicated nodepool or needs to be scheduled on a specific node,
the chart needs to support the value for `nodeSelector` or `affinity`.

This PR introduces both of those values.

I know this is too soon an implementation of issue https://github.com/redhat-actions/openshift-actions-runner-chart/issues/19 
But we can always close it or implement different where need be. 

### Related Issue(s)

- ability to schedule runner on specific node/nodepool

### Checklist

- [ ] This PR includes a documentation change
- [x] This PR does not need a documentation change
---
- [ ] This PR includes test changes
- [ ] This PR's changes are already tested
---
- [ ] This change is not user-facing
- [x] This change is a patch change
- [ ] This change is a minor change
- [ ] This change is a major (breaking) change

### Changes made

  - Change `deployment.yaml`  template adding `nodeSelector` and `affinity` specs
  - Update `values.yaml` adding `nodeSelector`, `affinity` and some links to the documentations

### CODE REVIEW

File: values.yaml
Comment (tetchel) [inline values.yaml]:
```suggestion
# Add nodeSelector to the deployment. This is easiest with a values file but can be done on the command line with:
```

File: values.yaml
Comment (tetchel) [inline values.yaml:59]: Can you move this line to the top of the comment block?

**Comment by tetchel** [comment]
@divyansh42 this is a good change, it just needs to be tested quickly by hand.

**Comment by koalalorenzo** [comment]
+1 for this feature :) 

@divyansh42 does it work? 🤞

File: values.yaml
Comment (emmanuelstroem) [reply values.yaml:59]: @tetchel 
the line highlight on Github is usable but could be more specific.
I suppose you mean move line 61 to 59. 🙈 

feel free to correct me there 🙏

Comment (emmanuelstroem) [reply values.yaml:59]: moved it to the top of the block 😄 
it does make the comment flow and example value uninterrupted.

**Review by divyansh42 — APPROVED** [review]
It's working totally fine.
Thanks :+1:

**Comment by koalalorenzo** [comment]
@divyansh42 @tetchel any blocker from merging this?

**Comment by divyansh42** [comment]
@tetchel could you please approve the changes?

**Comment by koalalorenzo** [comment]
@tetchel it would be nice to get another approval. We are blocked by this PR, and we would like to avoid using our own custom-managed helm chart to unlock ourselves 😅 Can you have a look and review it when you have some time soon? 🙏

**Comment by divyansh42** [comment]
@koalalorenzo I have created the release and the same is reflected in the `v1`.
Thanks @emmanuelstroem for your contribution :+1:

**Comment by tetchel** [comment]
Sorry all. I find myself in a difficult position with respect to maintaining this project (and others in this org). 

While I still feel a sense of ownership over this project since I was the original author, since I no longer work for Red Hat I cannot spend time to properly vet and test pull requests, plus I no longer have access to proper OpenShift clusters.

So while I can glance at a PR like this and be 99.9% sure it is totally fine, I do not feel comfortable formally approving it without checking it out and testing it, which I can no longer do. So, I want to defer to Divyanshu and others to merge PRs.

I probably should not have reviewed above, but I couldn't help myself.

Thanks for understanding.

**Comment by divyansh42** [comment]
@tetchel thanks for your time. The time you put into this is really appreciated. You don't need to be sorry at all. :slightly_smiling_face:

**Pull request merged**
